plugin/src/main/java/io/jenkins/plugins/casc/impl/secrets/VaultSecretSource.java

```diff
@@ -27,65 +27,124 @@
-            (vaultAppRole != null && vaultAppRoleSecret != null)) && vaultPth != null && vaultUrl != null) {
-            LOGGER.log(Level.FINE, "Attempting to connect to Vault: {0}", vaultUrl);
+
+        // attempt AppRole login
+        if (vaultAppRole.isPresent() && vaultAppRoleSecret.isPresent() && !authToken.isPresent()) {
             try {
-                VaultConfig config = new VaultConfig().address(vaultUrl);
-                if (vaultNamespace != null) {
-                    // optionally set namespace
-                    config = config.nameSpace(vaultNamespace);
-                    LOGGER.log(Level.FINE, "Using namespace with Vault: {0}", vaultNamespace);
-                }
-                if (vaultEngineVersion != null) {
-                    // optionally set vault engine version
-                    config = config.engineVersion( Integer.parseInt(vaultEngineVersion) );
-                    LOGGER.log(Level.FINE, "Using engine version: {0}", vaultEngineVersion);
-                }
-                config = config.build();
-                Vault vault = new Vault(config);
-                //Obtain a login token
-                final String token;
-                if (vaultToken != null) {
-                    token = vaultToken;
-                    LOGGER.log(Level.FINE, "Using supplied token to access Vault");
-                } else if (vaultAppRole != null && vaultAppRoleSecret != null) {
-                    token = vault.auth().loginByAppRole(vaultAppRole, vaultAppRoleSecret).getAuthClientToken();
-                    LOGGER.log(Level.FINE, "Login to Vault using AppRole/SecretID successful");
-                } else {
-                    token = vault.auth().loginByUserPass(vaultUsr, vaultPw, vaultMount).getAuthClientToken();
-                    LOGGER.log(Level.FINE, "Login to Vault using U/P successful");
+                authToken = Optional.of(
+                        vault.auth().loginByAppRole(vaultAppRole.get(), vaultAppRoleSecret.get()).getAuthClientToken()
+                );
+                LOGGER.log(Level.FINE, "Login to Vault using AppRole/SecretID successful");
+            } catch (VaultException e) {
+                LOGGER.log(Level.WARNING, "Could not login with AppRole", e);
+            }
+        }
+
+        // attempt User/Pass login
+        if (vaultUser.isPresent() && vaultPw.isPresent() && !authToken.isPresent()) {
+            try {
+                authToken = Optional.of(
+                        vault.auth().loginByUserPass(vaultUser.get(), vaultPw.get(), vaultMount.get()).getAuthClientToken()
+                );
+                LOGGER.log(Level.FINE, "Login to Vault using User/Pass successful");
+            } catch (VaultException e) {
+                LOGGER.log(Level.WARNING, "Could not login with User/Pass", e);
+            }
+        }
+
+        // Use authToken to read secrets from vault
+        readSecretsFromVault(authToken.get(), vaultConfig, vault, vaultPaths.get());
```

Comment:
missing check if authToken and vaultPaths are present

Reply:
Good point. I am actually now checking the vaultPaths as mandatory variable in the beginning. Still adding it here again. Better safe than sorry :+1: